system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -775,6 +789,352 @@ extern ICompiledXpath* compileXpath(const char * xpath)
+    {
+        bool ret = dft;
+        xmlXPathObjectPtr eval = xmlXPathEval((const xmlChar *) xpath, xpathCtx);
+        if (eval)
+        {
+            xmlChar *val = xmlXPathCastToString(eval);
+            xmlXPathFreeObject(eval);
+            if (val)
+            {
+                if (*val)
+                    ret = strToBool((const char *)val);
+                xmlFree(val);
+            }
+        }
+        return ret;
+    }
+
+    xmlNodePtr getSection(const char *name)
+    {
+        return getSectionNode(name, nullptr);
+    }
+    xmlNodePtr ensureSection(const char *name)
+    {
+        xmlNodePtr sect = getSection(name);
+        if (sect)
+            return sect;
+        return xmlNewChild(root, nullptr, (const xmlChar *) name, nullptr);
+    }
+    void removeSection(const char *name)
+    {
+        xmlXPathObjectPtr eval = xmlXPathEval((const xmlChar *) name, xpathCtx);
+        if (!eval)
+            return;
+        if (XPATH_NODESET==eval->type && !xmlXPathNodeSetIsEmpty(eval->nodesetval))
+        {
+            xmlNodeSetPtr ns = eval->nodesetval;
+            int count = xmlXPathNodeSetGetLength(ns);
+            for(int i=0; i<count; i++)
+            {
+                xmlNodePtr node = xmlXPathNodeSetItem(ns, i);
+                if (node)
+                {
+                    xmlUnlinkNode(node);
+                    xmlFreeNode(node);
+                    ns->nodeTab[i]=nullptr;
+                }
+            }
+        }
+        xmlXPathFreeObject(eval);
+    }
+    xmlNodePtr replaceSection(const char *name)
+    {
+        removeSection(name);
+        return xmlNewChild(root, nullptr, (const xmlChar *) name, nullptr);
+    }
+    virtual void setContent(const char *section, const char *xml) override
+    {
+        xmlNodePtr sect = replaceSection(section);
+        if (xml==nullptr) //means delete content
+            return;
+
+        xmlParserCtxtPtr parserCtx = xmlCreateDocParserCtxt((const unsigned char *)xml);
+        if (!parserCtx)
+            throw MakeStringException(-1, "CEsdlScriptContext:setXmlSection: Unable to init parse of %s XML content", section);
+//        parserCtx->myDoc = doc;
+        parserCtx->node = sect;
+        xmlParseDocument(parserCtx);
+        int wellFormed = parserCtx->wellFormed;
+        xmlFreeDoc(parserCtx->myDoc); //dummy document
+        xmlFreeParserCtxt(parserCtx);
+        if (!wellFormed)
+           throw MakeStringException(-1, "CEsdlScriptContext:setContent xml string: Unable to parse %s XML content", section);
+    }
+    virtual void setContent(const char *section, IPropertyTree *tree) override
+    {
+        xmlNodePtr sect = replaceSection(section);
+        if (tree==nullptr) //means delete content
+            return;
+        addChildFromPtree(sect, *tree, tree->queryName());
+    }
+
+    virtual void setAttribute(const char *section, const char *name, const char *value) override
```

Comment:
Maybe helpful looking toward the future, perhaps we should use some enumerations or constants for the section and name values used here. We might not want to *limit* it to an enumeration, but some standard constants would help with consistency.

Reply:
I agree we could use constants but they would probably be in the esdl binding.  The esdl script can handle any names, the binding sets our model for what our entry points actually are.  Although some may be common like "store" so could consider some being in esdl_script and some being in the esdl service/binding.